File: wlr/OGUR/OGUR/States/GameplayState.cs
using SPX.States;
using SPX.Entities;
using OGUR.Creatures;
using OGUR.Dungeons;
using SPX.Core;
using System;

namespace OGUR.States
{
    public class GameplayState : State
    {
        public GameplayState(OgurSave saveData = null)
        {
            EntityManager.Reset();
            CreatureFactory.ResetPlayerCount();
            if (saveData == null)
            {
                DungeonFactory.Start();
            }
            else
            {
                DungeonFactory.Start(saveData);
            }
            Console.WriteLine(Input.GetPlayerCount());
        }
        public override void Update()
        {
            EntityManager.Update();
        }
        public override void LoadContent()
        {
            EntityManager.LoadContent();
        }
        public override void Draw()
        {
            EntityManager.Draw();
        }
    }
}

File: code/aigilas-ported/src/com/aigilas/states/GameplayState.java
package com.aigilas.states;import com.xna.wrapper.*;import java.util.*;import com.spx.entities.*;import com.aigilas.creatures.*;import com.aigilas.dungeons.*;import com.spx.core.*;import com.spx.io.*;import com.spx.states.State;
    public class GameplayState  implements  State
    {
        public GameplayState()
        {
            Console.WriteLine("Generating the dungeon...");
            EntityManager.Reset();
            DungeonFactory.Start();
            Client.Get().DungeonHasLoaded();
        }
        public void Update()
        {
            EntityManager.Update();
        }
        public void LoadContent()
        {
            EntityManager.LoadContent();
        }
        public void Draw()
        {
            EntityManager.Draw();
        }
    }
